### PR TITLE
Cache PLTs in Github Actions

### LINKED ISSUE #157: CI optimisation - cache dialyxir plt filles
https://github.com/jeremyjh/dialyxir

https://github.com/CrowdHailer/Ace/blob/ab906826c55b87def4d6a1877cc39e995c8bb396/.github/workflows/elixir.yml#L37-L38

Something like 

http://stratus3d.com/blog/2020/05/01/running-dialyzer-in-jenkins-builds/

### PR DESCRIPTION
**Pull request by pablocostass**
Should close #157

### CODE REVIEW

**Comment by bryanhuntesl** [comment]
Nice @pablocostass

**Comment by bryanhuntesl** [comment]
@pablocostass  did you see in Cory’s one he generates the plt filles and catches them in two steps https://github.com/coryodaniel/k8s/blob/develop/.github/workflows/ci.yaml

**Comment by CrowdHailer** [comment]
Looks good, worth moving more to Cory's style?

**Comment by pablocostass** [comment]
@bryanhuntesl No I hadn't, I just learnt that you can do conditionals with the cache step output like that!

@CrowdHailer If you'd rather use that approach I can do that. The only difference, however, between Cory's config and mine is that his handles the creation of the directory where Dialyzer's PLT will be generated purely from the CI config file, whereas mine expects the directory to exist (i.e., it does not `mkdir` it and that's why it has a `.gitignore` to force the directory to exist).

In short, if you think it's cleaner to have a conditional step to generate the PLT for the first time and otherwise call Dialyzer to do its checks instead of simply calling Dialyzer, I'll do the change :)

**Comment by CrowdHailer** [comment]
I'm really not sure. I haven't looked into the details yet. Happy to go with which ever yu think best

**Comment by pablocostass** [comment]
I would leave it as is, that way Dialyzer is just called once in either scenario (i.e., when creating the PLT or when checking for warnings).

**Comment by CrowdHailer** [comment]
Done
